crm/main.py

```diff
@@ -0,0 +1,479 @@
+        data = []
+
+        if resource == 'all':
+            for res in ["ipv4_route", "ipv6_route", "ipv4_nexthop", "ipv6_nexthop", "ipv4_neighbor", "ipv6_neighbor",
+                        "nexthop_group_member", "nexthop_group_object", "acl_table", "acl_group", "acl_entry",
+                        "acl_counter", "fdb_entry"]:
+                data.append([res, crm_info[res + "_threshold_type"], crm_info[res + "_low_threshold"], crm_info[res + "_high_threshold"]])
+        else:
+            data.append([resource, crm_info[resource + "_threshold_type"], crm_info[resource + "_low_threshold"], crm_info[resource + "_high_threshold"]])
+
+        print '\n'
+        print tabulate(data, headers=header, tablefmt="simple", missingval="")
+        print '\n'
+
+    def show_resources(self, resource):
+        """
+        CRM Handler to display resources information.
+        """
+        countersdb = swsssdk.SonicV2Connector(host='127.0.0.1')
+        countersdb.connect(countersdb.COUNTERS_DB)
+
+        crm_stats = countersdb.get_all(countersdb.COUNTERS_DB, 'CRM:STATS')
+
+        header = ("Resource Name", "Used Count", "Available Count")
+        data = []
+
+        if resource == 'all':
+            for res in ["ipv4_route", "ipv6_route", "ipv4_nexthop", "ipv6_nexthop", "ipv4_neighbor", "ipv6_neighbor",
+                        "nexthop_group_member", "nexthop_group_object", "fdb_entry"]:
+                data.append([res, crm_stats['crm_stats_' + res + "_used"], crm_stats['crm_stats_' + res + "_available"]])
+        else:
+            data.append([resource, crm_stats['crm_stats_' + resource + "_used"], crm_stats['crm_stats_' + resource + "_available"]])
+
+        print '\n'
+        print tabulate(data, headers=header, tablefmt="simple", missingval="")
+        print '\n'
+
+    def show_acl_resources(self):
+        """
+        CRM Handler to display ACL recources information.
+        """
+        countersdb = swsssdk.SonicV2Connector(host='127.0.0.1')
+        countersdb.connect(countersdb.COUNTERS_DB)
+
+        header = ("Stage", "Bind Point", "Resource Name", "Used Count", "Available Count")
+        data = []
+
+        for stage in ["INGRESS", "EGRESS"]:
+            for bind_point in ["PORT", "LAG", "VLAN", "RIF", "SWITCH"]:
+                crm_stats = countersdb.get_all(countersdb.COUNTERS_DB, 'CRM:ACL_STATS:{0}:{1}'.format(stage, bind_point))
+
+                if crm_stats:
+                    for res in ["acl_group", "acl_table"]:
+                        data.append([
+                                        stage, bind_point, res,
+                                        crm_stats['crm_stats_' + res + "_used"],
+                                        crm_stats['crm_stats_' + res + "_available"]
+                                    ])
+
+        print '\n'
+        print tabulate(data, headers=header, tablefmt="simple", missingval="")
+        print '\n'
+
+    def show_acl_table_resources(self, table_id):
```

Comment:
Show table resource requires user to specify table_id? Just thinking if we can also display the complete table. What do you say?

Was this intended in design spec, `[resources|thresholds] acl group [entry|counter]`

Reply:
Currently it requires table_id, I agree that it is better to display complete table. Will check what we can do.